src/data/balance/pokemon-evolutions.ts

```diff
@@ -121,19 +123,233 @@ export class FusionSpeciesFormEvolution extends SpeciesFormEvolution {
 export class SpeciesEvolutionCondition {
   public predicate: EvolutionConditionPredicate;
   public enforceFunc: EvolutionConditionEnforceFunc | undefined;
+  public description: String;
 
   constructor(predicate: EvolutionConditionPredicate, enforceFunc?: EvolutionConditionEnforceFunc) {
     this.predicate = predicate;
     this.enforceFunc = enforceFunc;
+    this.description = "";
   }
 }
 
-export class SpeciesFriendshipEvolutionCondition extends SpeciesEvolutionCondition {
-  constructor(friendshipAmount: integer, predicate?: EvolutionConditionPredicate, enforceFunc?: EvolutionConditionEnforceFunc) {
-    super(p => p.friendship >= friendshipAmount && (!predicate || predicate(p)), enforceFunc);
+export class GenderEvolutionCondition extends SpeciesEvolutionCondition {
+  public gender: Gender;
+  constructor(gender: Gender) {
+    super(p => p.gender === gender, p => p.gender = gender);
+    this.gender = gender;
+    this.description = i18next.t("pokemonEvolutions:gender", { gender: i18next.t(`pokemonEvolutions:${Gender[gender]}`) });
+    // console.log("Description:", this.description);
   }
 }
 
+export class TimeOfDayEvolutionCondition extends SpeciesEvolutionCondition {
+  public timesOfDay: TimeOfDay[];
+  constructor(tod: string) {
+    if (tod === "day") {
+      super(p => globalScene.arena.getTimeOfDay() === TimeOfDay.DAWN || globalScene.arena.getTimeOfDay() === TimeOfDay.DAY);
+      this.timesOfDay = [ TimeOfDay.DAWN, TimeOfDay.DAY ];
+    } else if (tod === "night") {
+      super(p => globalScene.arena.getTimeOfDay() === TimeOfDay.DUSK || globalScene.arena.getTimeOfDay() === TimeOfDay.NIGHT);
+      this.timesOfDay = [ TimeOfDay.DUSK, TimeOfDay.NIGHT ];
+    } else {
+      super(p => false);
+      this.timesOfDay = [];
+    }
+    this.description = i18next.t("pokemonEvolutions:timeOfDay", { tod: i18next.t(`pokemonEvolutions:${tod}`) });
+    // console.log("Description:", this.description);
+  }
+}
+
+export class MoveEvolutionCondition extends SpeciesEvolutionCondition {
+  public move: Moves;
+  constructor(move: Moves) {
+    super(p => p.moveset.filter(m => m?.moveId === move).length > 0);
+    this.move = move;
+    const moveKey = Moves[this.move].split("_").filter(f => f).map((f, i) => i ? `${f[0]}${f.slice(1).toLowerCase()}` : f.toLowerCase()).join("") as unknown as string;
```

Comment:
```suggestion
    const moveKey = Moves[this.move].split("_").filter(f => f).map((f, i) => i ? `${f[0]}${f.slice(1).toLowerCase()}` : f.toLowerCase()).join("");
```
Why the `as unknown as string`? It seems to correctly identify that it's a string without it.
![image](https://github.com/user-attachments/assets/6e157d60-49dc-4294-ba4e-817450551b7a)

Reply:
I actually copied that from somewhere else in the code; perhaps it made sense there. Anyway I agree that it is not necessary.